File: Surya_Prakash/Day 3/Find whether an array is a subset of another array.py
for t in range(int(input())):
    m, n = map(int, input().split())
    arr1 = list(map(int, input().split()))
    arr2 = list(map(int, input().split()))
    mp = {}
    ans = 'Yes'
    for i in arr1:
        if mp.get(i, False) == False:
            mp[i] = True

    for i in arr2:
        if mp.get(i, False) == False:
            ans = 'No'
            break
    print(ans)

File: parasmittal/Day 3/Find whether an array is a subset of another array.cpp
#include <iostream>
using namespace std;
#include<bits/stdc++.h>
int main() {
	//code
	int t;
	cin>>t;
	while(t--){
	    int m,n;
	    cin>>m>>n;
	    int arr1[m],arr2[n];
	    for(int i=0;i<m;i++)
	    cin>>arr1[i];
	    for(int i=0;i<n;i++)
	    cin>>arr2[i];
	    if(n>m){
	        cout<<"No"<<endl;
	        continue;
	    }
	    unordered_map<int,int> um;
	    for(int i=0;i<m;i++)
	    um[arr1[i]]++;
	    bool flag=true;
	    for(int i=0;i<n;i++){
	        if(um.find(arr2[i])==um.end()){
	            flag=false;
	            break;
	        }
	        um[arr2[i]]--;
	    }
	    if(!flag){
	        cout<<"No"<<endl;
	    }
	    else
	    cout<<"Yes"<<endl;
	}
	return 0;
}